File: backend/src/main/java/com/semitris/acg/entity/Collection.java
package com.semitris.acg.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.apache.ibatis.type.Alias;

import java.time.LocalDateTime;

/**
 * @ClassName Collection
 * @Description 收藏表实体类
 * @Author SemiTris
 * @Date 2026年08月27日 21:14
 * @Version 1.0
 */

@Data
@AllArgsConstructor
@NoArgsConstructor
@Alias("userCollection")
public class Collection {

    private Long id;                //主键
    private Long userId;            //用户ID
    private Long animeId;           //番剧ID
    private Integer listType;       //收藏类型：1=想看 2=在看 3=已看完 4=搁置 5=最爱
    private Integer progress;       //看到第几集/话
    private Integer favorite;       //是否加精收藏：0=否 1=是
    private LocalDateTime createdAt;//收藏时间
    private LocalDateTime updatedAt;//更新时间

    private String title;           //番剧标题（连表非持久化字段，不入库）
    private String cover;           //番剧封面URL（连表非持久化字段，不入库）

}

File: frontend/src/api/collection.js
import request from '@/utils/request'

// 追番列表（listType 可选：1~5，不传=全部；连表 title/cover）
export function listCollection(listType) {
  return request.get('/collection/list', { params: { listType: listType || undefined } })
}

// 按番查询收藏状态（详情页判断；未收藏 data=null）
export function findByAnime(animeId) {
  return request.get(`/collection/findByAnime/${animeId}`)
}

// 新增收藏（userId/时间不传，后端填充；listType 1~5、favorite 0/1、progress<=episodes）
export function addCollection(data) {
  return request.post('/collection/add', data)
}

// 更新收藏（只传变更字段 + id；listType/progress/favorite）
export function updateCollection(data) {
  return request.put('/collection/update', data)
}

// 取消收藏
export function deleteCollection(id) {
  return request.delete(`/collection/delete/${id}`)
}
